core/templates/domain/skill/skill-backend-api.service.spec.ts

```diff
@@ -118,165 +111,208 @@ describe('Skill backend API service', function() {
-      $httpBackend.flush();
+      var req = httpTestingController.expectOne(
+        '/skill_editor_handler/data/1');
+      expect(req.request.method).toEqual('GET');
+      req.flush(sampleResponse);
+
+      flushMicrotasks();
 
       expect(successHandler).toHaveBeenCalledWith({
         skill: sampleResponse.skill,
-        groupedSkillSummaries: sampleResponse.grouped_skill_summaries
+        groupedSkillSummaries: sampleResponse.groupedSkillSummaries
       });
       expect(failHandler).not.toHaveBeenCalled();
-    });
+    }));
 
   it('should use the rejection handler if backend request failed',
-    function() {
-      var successHandler = jasmine.createSpy('success');
-      var failHandler = jasmine.createSpy('fail');
+    fakeAsync(() => {
+      let successHandler = jasmine.createSpy('success');
+      let failHandler = jasmine.createSpy('fail');
 
-      $httpBackend.expect('GET', '/skill_editor_handler/data/1').respond(
-        500, 'Error loading skill 1.');
-      SkillBackendApiService.fetchSkill('1').then(
+      skillBackendApiService.fetchSkill('1').then(
         successHandler, failHandler);
-      $httpBackend.flush();
+      var req = httpTestingController.expectOne(
+        '/skill_editor_handler/data/1');
+      expect(req.request.method).toEqual('GET');
+      req.flush('Error loading skill 1.', {
+        status: 500,
+        statusText: 'Error loading skill 1.'
+      });
+
+      flushMicrotasks();
 
       expect(successHandler).not.toHaveBeenCalled();
       expect(failHandler).toHaveBeenCalledWith('Error loading skill 1.');
-    });
+    }));
 
   it('should make a request to update the skill in the backend',
-    function() {
-      var successHandler = jasmine.createSpy('success');
-      var failHandler = jasmine.createSpy('fail');
-
-      $httpBackend.expect('GET', '/skill_editor_handler/data/1').respond(
-        sampleResponse);
-
-      var skillDict = null;
-      SkillBackendApiService.fetchSkill('1').then(
-        function(data) {
-          skillDict = data.skill;
+    fakeAsync(() => {
+      let successHandler = jasmine.createSpy('success');
+      let failHandler = jasmine.createSpy('fail');
+
+      let skillDict = null;
+      skillBackendApiService.fetchSkill('1').then(
+        (data) => {
+          skillDict = data;
         });
-      $httpBackend.flush();
+      var reqGet = httpTestingController.expectOne(
+        '/skill_editor_handler/data/1');
+      expect(reqGet.request.method).toEqual('GET');
+      reqGet.flush(sampleResponse);
+
+      flushMicrotasks();
 
-      $httpBackend.expect('PUT', '/skill_editor_handler/data/1').respond({
+      skillBackendApiService.updateSkill(
+        skillDict.skill.id, skillDict.version, 'commit message', []).then(
+        successHandler, failHandler);
+      var reqPut = httpTestingController.expectOne(
+        '/skill_editor_handler/data/1');
+      expect(reqPut.request.method).toEqual('PUT');
+      reqPut.flush({
```

Comment:
Could we write this in one line rather than breaking at the opening bracket?

Reply:
Done!